code/game/objects/structures/stool_bed_chair_nest/chairs.dm

```diff
@@ -139,40 +139,44 @@
 	color = rgb(255,251,0)
 
 /obj/structure/bed/chair/withwheels
+	var/delay = 10
 	var/cooldown = 0
+	var/reverse_direction = 1
 
-/obj/structure/bed/chair/withwheels/relaymove(mob/user, direction)
+/obj/structure/bed/chair/withwheels/relaymove(mob/user, direction, delay=10)
 	if((!Process_Spacemove(direction)) || (!has_gravity(src.loc)) || (cooldown) || user.stat || user.stunned || user.weakened || user.paralysis || (user.restrained()))
 		return
 	step(src, direction)
 	if(buckled_mob)
-		buckled_mob.dir = dir
-		switch(buckled_mob.dir)
-			if(NORTH)
-				buckled_mob.dir = SOUTH
-			if(WEST)
-				buckled_mob.dir = EAST
-			if(SOUTH)
-				buckled_mob.dir = NORTH
-			if(EAST)
-				buckled_mob.dir = WEST
+		buckled_mob.dir = direction
+		if(src.reverse_direction)
```

Comment:
using src. is most of the time useless, unless you have a case of an argument having the same name of a variable(like your current delay var/arg), just saying. It'd be better to remove that src.